src/patchelf.cc

```diff
@@ -1021,7 +1021,7 @@ void ElfFile<ElfFileParamNames>::rewriteHeaders(Elf_Addr phdrAddress)
                 if (shdr) {
                     auto rld_map_addr = findSectionHeader(".rld_map").sh_addr;
                     auto dyn_offset = ((char*)dyn) - ((char*)dyn_table);
-                    dyn->d_un.d_ptr = rld_map_addr + dyn_offset - (*shdrDynamic).get().sh_addr;
+                    dyn->d_un.d_ptr = rld_map_addr - dyn_offset - (*shdrDynamic).get().sh_addr;
```

Comment:
So the user of this pointer does `dyn->d_un.d_ptr + &dyn->d_un.d_ptr` to compute the debug pointer location?

Reply:
> So the user of this pointer does `dyn->d_un.d_ptr + &dyn->d_un.d_ptr` to compute the debug pointer location?

I'd say, it's `dyn + dyn->d_un.d_ptr`. Here is, for example, the relevant part of the glibc's elf loader code (`sysdeps/mips/dl-machine.h`):

```
char *ptr = (char *)(l)->l_info[DT_MIPS (RLD_MAP_REL)]; \
ptr += (l)->l_info[DT_MIPS (RLD_MAP_REL)]->d_un.d_val; \
```